### PR TITLE
[WIP] runner: check image exists locally

### PR DESCRIPTION
**Pull request by pedronasser**

### CODE REVIEW

**Comment by ccirello** [comment]
@nikhilm what you say of this PR? Shouldn't this be part of iron-io/runner perhaps?

**Comment by denismakogon** [comment]
Agreed with @ccirello seems like a part of runner lib. And there's obvious reason for this, all container-type specific API is encapsulated in runner to provide high-level abstraction. So, having explicit requests to Docker API would bring certain inconsistency.

File: api/runner/docker/client.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/docker/docker/client"
+	"golang.org/x/net/context"
+)
+
+func CheckImageLocally(image string) error {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		panic(err)
+	}
+	// Have to use ctx.Background() for now because the docker official client
```
Comment (nikhilm) [inline api/runner/docker/client.go]: Our common.Context has nothing to do with context.Context.

**Comment by pedronasser** [comment]
@nikhilm 
The idea of the PR is to check if the image exists locally.
Right now we are checking only against the registry.
`driver.CanExecuteFast`, would serve for that case?

**Comment by nikhilm** [comment]
You would have to write that function ;)

Best,
Nikhil




On Wed, Dec 7, 2016 at 2:19 PM -0800, "Pedro Nasser" <notifications@github.com> wrote:












@nikhilm

The idea of the PR is to check if the image exists locally.

Right now we are checking only against the registry.

driver.CanExecuteFast, would serve for that case?

**Comment by pedronasser** [comment]
@nikhilm Look that PR https://github.com/iron-io/runner/pull/24

**Pull request closed without merging**